test/new-e2e/tests/orchestrator/k8s_api_key_test.go

```diff
@@ -0,0 +1,107 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package orchestrator
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	awskubernetes "github.com/DataDog/datadog-agent/test/new-e2e/pkg/provisioners/aws/kubernetes"
+	"github.com/DataDog/test-infra-definitions/components/datadog/kubernetesagentparams"
+)
+
+//go:embed agent_api_key_values.yaml
+var agentAPIKeyRefreshValuesFmt string
+
+// TestZZZClusterAgentAPIKeyRefresh tests the agent's ability to refresh the API key
+// ZZZ is used to ensure this test runs last in the suite as it requires redeploying the agent
+func (suite *k8sSuite) TestZZZClusterAgentAPIKeyRefresh() {
```

Comment:
```suggestion
func (suite *k8sSuite) TestZzzClusterAgentAPIKeyRefresh() {
```

Since 'z' > 'Z' in ASCII

Reply:
And please adjust the comment

Reply:
thanks for the review: https://github.com/DataDog/datadog-agent/pull/35815/commits/d4d1f03d5075a210c97b2becc52333879481a0fe